python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -967,9 +968,12 @@ def my_dbt_op(dbt: DbtCliResource):
                     dbt_macro_args = {"key": "value"}
                     dbt.cli(["run-operation", "my-macro", json.dumps(dbt_macro_args)]).wait()
         """
+        op_context = (
+            context.op_execution_context if isinstance(context, AssetExecutionContext) else context
+        )
```

Comment:
Would prefer that we just keep this named `context`. 

Can we move this to be after we instantiate `assets_def`, but before the `if context and assets_def is not None`?

```suggestion
        context: Optional[OpExecutionContext] = (
            context.op_execution_context if isinstance(context, AssetExecutionContext) else context
        )
```

Reply:
sure for the rename, but `_get_unique_target_name` needs the op context too and is called before we get the assets_def. I just moved fetching `assets_def` up, but lmk if you'd prefer something else